src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -732,6 +732,22 @@ public SolutionState WithProjectDefaultNamespace(ProjectId projectId, string? de
             return ForkProject(newProject);
         }
 
+        /// <summary>
+        /// Creates a new solution instance with the project specified updated to have the name.
+        /// </summary>
+        public SolutionState WithProjectChecksumAlgorithm(ProjectId projectId, SourceHashAlgorithm checksumAlgorithm)
+        {
+            var oldProject = GetRequiredProjectState(projectId);
+            var newProject = oldProject.WithChecksumAlgorithm(checksumAlgorithm);
+
+            if (oldProject == newProject)
+            {
+                return this;
+            }
+
+            return ForkProject(newProject);
```

Comment:
Doesn't this need to include a CompilationTranslationAction to replace the syntax trees in the compilation with updated ones that match the algorithm? (This implies a missing test somewhere...)

Reply:
This is very much a blocking issue as this could mean stuff doesn't actually work how you want it to.

Reply:
Good catch.

Reply:
This is still missing?

Reply:
Added.